controlplane/kubeadm/internal/workload_cluster_etcd_test.go

```diff
@@ -491,26 +494,70 @@ kind: ClusterStatus`,
 
 	tests := []struct {
 		name                string
+		kubernetesVersion   semver.Version
 		objs                []client.Object
 		nodes               []string
 		etcdClientGenerator etcdClientFor
 		expectErr           bool
-		assert              func(*WithT)
+		assert              func(*WithT, client.Client)
 	}{
 		{
 			// the node to be removed is ip-10-0-0-3.ec2.internal since the
 			// other two have nodes
-			name:  "successfully removes the etcd member without a node and removes the node from kubeadm config",
-			objs:  []client.Object{node1.DeepCopy(), node2.DeepCopy(), kubeadmConfig.DeepCopy()},
-			nodes: []string{node1.Name, node2.Name},
+			name:              "successfully removes the etcd member without a node and removes the node from kubeadm config for Kubernetes version < 1.22.0",
+			kubernetesVersion: semver.MustParse("1.19.1"), // Kubernetes version < 1.22.0 has ClusterStatus
+			objs:              []client.Object{node1.DeepCopy(), node2.DeepCopy(), kubeadmConfig.DeepCopy()},
+			nodes:             []string{node1.Name, node2.Name},
+			etcdClientGenerator: &fakeEtcdClientGenerator{
+				forNodesClient: &etcd.Client{
+					EtcdClient: fakeEtcdClient,
+				},
+			},
+			expectErr: false,
+			assert: func(g *WithT, c client.Client) {
+				g.Expect(fakeEtcdClient.RemovedMember).To(Equal(uint64(3)))
+
+				var actualConfig corev1.ConfigMap
+				g.Expect(c.Get(
+					ctx,
+					client.ObjectKey{Name: kubeadmConfigKey, Namespace: metav1.NamespaceSystem},
+					&actualConfig,
+				)).To(Succeed())
+
+				g.Expect(actualConfig.Data[clusterStatusKey]).To(Equal("apiEndpoints:\n" +
```

Comment:
Could it be worth to switch to multiline strings here?

Reply:
IMO multiline strings makes the code less readable when using yaml, because the indentation gets abrutly interrupted.
Also, we already did this for recent changes in KCP (e.g. all the refactor on the kubeadm config update)...

Reply:
Could you provide an example? I'm not sure how this is better than a straight multiline string

Reply:
I'm happy to change if you feel this is better, but here it is an example of indentation hard to read due to multiline strings

https://github.com/kubernetes-sigs/cluster-api/blob/7478817225e0a75acb6e14fc7b438231578073d2/controlplane/kubeadm/internal/workload_cluster_test.go#L809-L821

This is ( much more complex) an example without multiline strings; indentation is consisted and easier to read

https://github.com/kubernetes-sigs/cluster-api/blob/5c19ea1f6099e68539146aa40a7631bda618c6c1/controlplane/kubeadm/internal/workload_cluster_test.go#L986-L1018

Reply:
You could start the multi-line string with a new line and use TrimSpace https://play.golang.org/p/YShA372kY82 ?

Reply:
Let's tackle this in a new PR?